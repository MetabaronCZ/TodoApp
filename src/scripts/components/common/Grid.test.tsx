import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { Grid } from 'components/common/Grid';
import { withMockedProviders } from 'test/component';

describe('components/common/Grid', () => {
  it('should render correctly', () => {
    const tree = render(withMockedProviders(<Grid>CONTENT</Grid>));
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByText('CONTENT')).toBeInTheDocument();
  });

  it('should render props', () => {
    const tree = render(withMockedProviders(<Grid size={4} />));
    expect(tree.container).toMatchSnapshot();
    expect(tree.container.firstChild).toHaveStyle({ gap: '40px' });
  });

  it('should keep minimum prop size', () => {
    const tree = render(withMockedProviders(<Grid size={-2} />));
    expect(tree.container.firstChild).toHaveStyle({ gap: '0px' });
  });
});
