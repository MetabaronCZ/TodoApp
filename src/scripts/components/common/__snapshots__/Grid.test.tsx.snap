// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/common/Grid > should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 20px;
}

<div>
  <div
    class="c0"
  >
    CONTENT
  </div>
</div>
`;

exports[`components/common/Grid > should render props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 40px;
}

<div>
  <div
    class="c0"
  />
</div>
`;
